atlasdb-cassandra/src/test/java/com/palantir/atlasdb/cassandra/CassandraReloadableKeyValueServiceRuntimeConfigTest.java

```diff
@@ -0,0 +1,225 @@
+                CassandraReloadableKeyValueServiceRuntimeConfig.fromConfigs(
+                        config, Refreshable.only(RUNTIME_CONFIG_WITH_DEFAULT_SERVERS));
+
+        assertThat(reloadableConfig.get().concurrentGetRangesThreadPoolSize()).isEqualTo(1234);
+    }
+
+    @Test
+    public void mergedConfigDerivesThreadPoolSizeIfInstallIsEmpty() {
+        int poolSize = 4321;
+        CassandraKeyValueServiceConfig config = configBuilderWithDefaultCredentialsAndReplicationFactor()
+                .poolSize(poolSize)
+                .build();
+        Refreshable<CassandraReloadableKeyValueServiceRuntimeConfig> reloadableConfig =
+                CassandraReloadableKeyValueServiceRuntimeConfig.fromConfigs(
+                        config, Refreshable.only(RUNTIME_CONFIG_WITH_DEFAULT_SERVERS));
+
+        assertThat(reloadableConfig.get().concurrentGetRangesThreadPoolSize())
+                .isEqualTo(poolSize * SERVERS_1.numberOfThriftHosts());
+    }
+
+    @Test
+    public void mergedConfigPrioritisesInstallForDefaultGetRangesConcurrency() {
+        CassandraKeyValueServiceConfig config = configBuilderWithDefaultCredentialsAndReplicationFactor()
+                .concurrentGetRangesThreadPoolSize(1243)
+                .build();
+
+        Refreshable<CassandraReloadableKeyValueServiceRuntimeConfig> reloadableConfig =
+                CassandraReloadableKeyValueServiceRuntimeConfig.fromConfigs(
+                        config, Refreshable.only(RUNTIME_CONFIG_WITH_DEFAULT_SERVERS));
+
+        assertThat(reloadableConfig.get().defaultGetRangesConcurrency())
+                .isEqualTo(config.defaultGetRangesConcurrency());
+    }
+
+    @Test
+    public void mergedConfigDerivesDefaultGetRangesConcurrencyIfInstallIsEmpty() {
+        CassandraKeyValueServiceConfig config = configBuilderWithDefaultCredentialsAndReplicationFactor()
+                .poolSize(10)
+                .build();
+
+        Refreshable<CassandraReloadableKeyValueServiceRuntimeConfig> reloadableConfig =
+                CassandraReloadableKeyValueServiceRuntimeConfig.fromConfigs(
+                        config, Refreshable.only(RUNTIME_CONFIG_WITH_DEFAULT_SERVERS));
+
+        assertThat(reloadableConfig.get().defaultGetRangesConcurrency()).isEqualTo(8);
+    }
+
+    private static ImmutableCassandraKeyValueServiceConfig.Builder
+            configBuilderWithDefaultCredentialsAndReplicationFactor() {
+        return ImmutableCassandraKeyValueServiceConfig.builder()
+                .credentials(CREDENTIALS)
+                .replicationFactor(1);
+    }
+
+    private static ImmutableCassandraKeyValueServiceRuntimeConfig.Builder runtimeConfigBuilder() {
+        return ImmutableCassandraKeyValueServiceRuntimeConfig.builder();
+    }
+
+    private static ImmutableCassandraKeyValueServiceRuntimeConfig.Builder runtimeConfigBuilderWithDefaultServers() {
+        return runtimeConfigBuilder().servers(SERVERS_1);
+    }
```

Comment:
do we need to test checkSharedGetRangesPoolGreaterThanOrEqualToConcurrentGetRangesThreadPool in some way here?

Reply:
Thanks for the flag, I dropped it when rewriting some of the tests perhaps too aggressively.